notary-plugins/notary-plugin-non-validating/notary-plugin-non-validating-server/src/main/kotlin/com/r3/corda/notary/plugin/nonvalidating/server/NonValidatingNotaryServerFlowImpl.kt

```diff
@@ -111,6 +122,21 @@ class NonValidatingNotaryServerFlowImpl() : ResponderFlow {
         }
     }
 
+    /**
+     * This function will validate selected notary is valid notary to notarize.
+     * */
+    @Suspendable
+    private fun getCurrentNotaryAndValidateNotary(txDetails: NonValidatingNotaryTransactionDetails) {
+        val currentNotaryMemberProvidedCtx = memberLookup.myInfo().memberProvidedContext
+        val currentNotaryServiceName = currentNotaryMemberProvidedCtx.parse(NOTARY_SERVICE_NAME, MemberX500Name::class.java)
+
+        val payloadNotaryServiceName = txDetails.notaryName
+
+        require(currentNotaryServiceName == payloadNotaryServiceName) {
```

Comment:
```suggestion
        val currentNotaryServiceName = memberLookup
            .myInfo()
            .memberProvidedContext
            .parse(NOTARY_SERVICE_NAME, MemberX500Name::class.java)

        require(currentNotaryServiceName == txDetails.notaryName) {
```

Removes two unnecessary temporary variables that are only used once, and aren't adding much value imo.